src/main/java/org/elasticsearch/index/translog/fs/FsChannelSnapshot.java

```diff
@@ -16,138 +16,58 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-
 package org.elasticsearch.index.translog.fs;
 
-import org.apache.lucene.util.BytesRef;
+import org.apache.lucene.util.IOUtils;
 import org.elasticsearch.ElasticsearchException;
-import org.elasticsearch.common.bytes.BytesArray;
-import org.elasticsearch.common.io.Channels;
-import org.elasticsearch.common.io.stream.BytesStreamInput;
-import org.elasticsearch.index.translog.TranslogStreams;
 import org.elasticsearch.index.translog.Translog;
 
-import java.io.EOFException;
-import java.io.FileNotFoundException;
+import java.io.Closeable;
 import java.io.IOException;
 import java.nio.ByteBuffer;
-import java.nio.channels.FileChannel;
 import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicLong;
 
 /**
- *
+ * an implementation of {@link org.elasticsearch.index.translog.Translog.Snapshot}, wrapping
+ * a {@link FsChannelReader}. This class is intended for a single thread use.
```

Comment:
"This class is intended for a single thread use." - can we change this to a more aggressive "This class is NOT thread-safe"? I want to make sure there is no confusion in the future for any other devs using this class